diff --git a/lib.rs b/lib.rs
index cee3870c629..3cf7fa0e4e5 100644
--- a/lib.rs
+++ b/lib.rs
@@ -278,6 +278,14 @@
 #![feature(wasm_target_feature)]
 // tidy-alphabetical-end
 
+//
+// Leaf-required features:
+// tidy-alphabetical-start
+#![feature(concat_idents)]
+#![feature(register_tool)]
+#![register_tool(leaf_attr)]
+// tidy-alphabetical-end
+
 // allow using `core::` in intra-doc links
 #[allow(unused_extern_crates)]
 extern crate self as core;
@@ -477,3 +485,6 @@ pub mod simd {
 }
 
 include!("primitive_docs.rs");
+
+#[cfg_attr(core_build, stable(feature = "rust1", since = "1.0.0"))]
+pub mod leaf;
